Esri/arcgis-runtime-samples-ios
# Update: Add filter barrier to Perform valve isolation trace

## Description

This PR updates the `Perform valve isolation trace` sample for `common-samples/issues/2354`, that adds the capability to tap on map and add filter barriers in the network, in addition to specifying a category comparison to run the trace.

## How to test

- The original workflow using category comparison should still work
- The new workflow of adding filter barrier in the network will give different results
- The 2 workflows cannot exist at the same time: you can either specify a category comparison, or add points to the map. This is achieved by changing UI states and checking nil for `selectedCategory` property.

Since there are lots of changes in UI states and functionalities, please also do extensive holistic ad-hoc testing to see if any UI logic is wrong or it crashes.
